pkg/liqonet/ipam/ipam.go

```diff
@@ -458,22 +457,24 @@ func (liqoIPAM *IPAM) freePoolInHalves(pool string) error {
 		return err
 	}
 
-	klog.Infof("Network %s is equal to a network pool, freeing first half..", pool)
-	err = liqoIPAM.ipam.ReleaseChildPrefix(liqoIPAM.ipam.PrefixFrom(halfCidr))
-	if err != nil {
-		return fmt.Errorf("cannot free first half of pool %s", pool)
+	klog.Infof("Network %s is equal to a network pool, freeing it in two steps...", pool)
+	for i := 1; i <= 2; i++ {
+		if i == 2 {
+			// Get second half CIDR
+			halfCidr, err = utils.Next(halfCidr)
+			if err != nil {
+				return err
+			}
+		}
+		klog.Infof("freeing half %d...", i)
+		if prefix := liqoIPAM.ipam.PrefixFrom(halfCidr); prefix != nil {
+			err = liqoIPAM.ipam.ReleaseChildPrefix(prefix)
+			if err != nil {
+				return fmt.Errorf("cannot free first half of pool %s: %w", pool, err)
+			}
+		}
```

Comment:
Here we are completely in the field of opinions, but I personally felt it clearer and more intuitive the previous unrolled version. Also in terms of overall length they seem similar to me. 
BTW, if the new version is maintained, the error message should be generalized.